lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -1463,8 +1459,9 @@ internal class StaticTypeInferenceVisitorTransform(
                 is ListType,
                 is SexpType -> {
                     val previous = previousComponentType as CollectionType // help Kotlin's type inference to be more specific
-                    val staticType = currentPathComponent.index.getStaticType()
-                    if (staticType is StaticScalarType && (staticType.scalarType in listOf(Int2Type, Int4Type, Int8Type, IntType))) {
+                    val staticType = currentPathComponent.index.getStaticType() as SingleType
+                    // As long as the expression surrounded by brackets can be evaluated as an integer scalar, we don't care which type it is at compile time
```

Comment:
Here, "brackets" mean the bracket notation we use to access an item in a list/sexp, e.g. `sexp(1, 2, 3)[0]`

Reply:
It's worth folding this explanation into the code comment.